scheduler/scheduler-examples/src/main/java/org/ow2/proactive/scheduler/examples/SFTPConnector.java

```diff
@@ -0,0 +1,378 @@
+                    throw new IllegalArgumentException("SFTP mode can only be PUT or GET.");
+            }
+        } catch (JSchException e) {
+            throw new JSchException("An error occured while trying to connect to SFTP server: ", e);
+        } catch (SftpException e) {
+            throw new SftpException(e.id,
+                                    "File not found, permission denied or operation not supported by SFTP server.",
+                                    e);
+        } finally {
+            if (channelsftp != null)
+                channelsftp.disconnect();
+            if (channel != null)
+                channel.disconnect();
+            if (session != null)
+                session.disconnect();
+            getOut().println("Disconnected from SFTP server");
+        }
+
+        return (Serializable) filesRelativePathName;
+    }
+
+    /**
+     * This method recursively uploads a local file/folder to an SFTP server
+     *
+     * @param sourcePath
+     * @param destinationPath
+     * @throws SftpException
+     * @throws FileNotFoundException
+     */
+    private List<String> recursiveFolderUpload(ChannelSftp channelSftp, String sourcePath, String destinationPath)
+            throws SftpException, FileNotFoundException {
+
+        List<String> filesRelativePathName = new ArrayList<>();
+        File sourceFile = new File(sourcePath);
+        String remoteFilePath = Paths.get(destinationPath, sourceFile.getName()).toString();
+
+        if (sourceFile.isFile()) {
+            // copy if it is a file
+            getOut().println("Uploading " + sourcePath);
+            channelSftp.put(new FileInputStream(sourceFile), remoteFilePath, ChannelSftp.OVERWRITE);
+            getOut().println("File uploaded successfully to " + remoteFilePath);
+            filesRelativePathName.add(remoteFilePath);
+
+        } else {
+
+            File[] files = sourceFile.listFiles();
+
+            if (files != null && !sourceFile.isHidden()) {
+
+                // create a directory if it does not exist
+                if (!remoteDirExists(channelSftp, remoteFilePath)) {
+                    channelSftp.mkdir(remoteFilePath);
+                }
+
+                for (File f : files) {
+                    filesRelativePathName.addAll(recursiveFolderUpload(channelSftp,
+                                                                       f.getAbsolutePath(),
+                                                                       remoteFilePath));
+                }
+            }
+        }
+        return filesRelativePathName;
+    }
+
+    /**
+     * This method downloads a single file over SFTP.<br>
+     * By default, this method overwrites existing files.
+     *
+     * @param channelSftp
+     * @param remoteFilePath
+     * @param localFilePath
+     * @return
+     * @throws SftpException
+     */
+    private String downloadSingleFile(ChannelSftp channelSftp, String remoteFilePath, String localFilePath)
```

Comment:
To reduce the scope of this class (that could be to only handle the SFTP connection), I suggest you could create two utility classes next to `SFTPConnection`. For example `SFTPConnectorDownloader` and `SFTPConnectorUploader`.